drivers/counter/counter_mcux_pit.c

```diff
@@ -130,17 +151,24 @@ static uint32_t mcux_pit_get_frequency(const struct device *dev)
 	return clock_rate;
 }
 
+
 static void mcux_pit_isr(const struct device *dev)
 {
 	const struct mcux_pit_config *config = dev->config;
-	struct mcux_pit_data *data = dev->data;
 	uint32_t flags;
 
 	LOG_DBG("pit counter isr");
-	flags = PIT_GetStatusFlags(config->base, config->pit_channel);
-	PIT_ClearStatusFlags(config->base, config->pit_channel, flags);
-	if (data->top_callback) {
-		data->top_callback(dev, data->top_user_data);
+
+	for (int channel_index = 0;
```

Comment:
@manuargue it looks like in the init macro all the interrupts are being connected to this routine, and this routine checks the peripheral register for which channels interrupted, so what did you mean is the issue?

Reply:
@manuargue you are right, but previously this driver didn't even support multiple channels, so I would like to ask is it necessary for author to implement this right now to merge

Reply:
Okay, now I agree that this is not an optimization, and it should be done in this PR, because of the contract of the interrupt priorities in devicetree.

Reply:
@manuargue Okay, I agree now that this is not an optimization because it is required by the contract of interrupts in DT. So, if there is an SOC that has the pit interrupts like this, then it should be done this way. But, I don't see an SOC in the tree that has multiple interrupts like this. Do you know of an SOC that does? Theoretically this argument could be made for any driver in zephyr, and most drivers do not have instance-based ISRs

Reply:
Thank you I missed this, so, I agree this is currently a bug in the PR for the k6x series

Reply:
I am still working on this.

Reply:
Both case scenarios of the interrupts are handled now with the small caveat of have conditional #ifdefs due to the way we are currently connecting the array of IRQs with its corresponding channel data device.